### PR TITLE
lang/web: Support go & hugo engines in web-mode

### PR DESCRIPTION
**Pull request by sei40kr**
Set an appropriate engine when editing a Go template or Hugo template in web-mode.

### CODE REVIEW

File: modules/lang/web/+html.el [hunk 1]
```diff
@@ -36,6 +36,13 @@
       ("-xml" (memq major-mode '(nxml-mode xml-mode))))
     :ok-statuses '(0 1))
 
+  (set-formatter! '+web-go-template
+    #'(format-all--buffer-native 'web-mode
+                                 (lambda ()
+                                   (indent-region (point-min) (point-max))))
```
Comment (sei40kr) [inline modules/lang/web/+html.el:42]: For now, there's no perfect formatter for Go templates. I tried Prettier & prettier-plugin-go-template, but it often breaks syntax.

File: modules/lang/web/autoload/html.el [hunk 2]
```diff
@@ -132,3 +132,11 @@ snippet, or `emmet-expand-yas'/`emmet-expand-line', depending on whether
               #'yas-expand
             #'emmet-expand-yas))
          (#'emmet-expand-line))))
+
+;;;###autoload
+(defun +web--go-template-format-buffer ()
```
Comment (sei40kr) [inline modules/lang/web/autoload/html.el:137]: I tried to implement this as `lambda` in `set-formatter!` but I couldn't make it work. So I implemented an interactive command instead.

**Comment by sei40kr** [comment]
@hlissner Could you review this PR?

**Comment by hlissner** [comment]
I'm closing this for the same reason #3011 was closed; I am not rejecting these changes, but pending changes to `:editor format` (in #6369) need to be accommodated for, and those efforts will happen in the [new module library](https://github.com/doomemacs/modules) repo instead of here. I will revisit it once these things have been sorted out. Thanks again!

**Pull request closed without merging**